core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -328,6 +356,18 @@ public void append(FieldStats stats) {
         }
 
         @Override
+        public String stringValueOf(Object value, String optionalFormat) {
+            if (optionalFormat != null) {
+                throw new UnsupportedOperationException("custom format isn't supported");
+            }
+            if (value instanceof Number) {
+                return java.lang.Float.toString(((Number) value).longValue());
```

Comment:
hmm this should be `((Number) value).floatValue()` no? I think we need better dedicated tests for all of those?